hibernate-core/src/main/java/org/hibernate/hql/spi/id/cte/ValuesListCteBuilder.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later.
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.hql.spi.id.cte;
+
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * @author Evandro Pires da Silva
+ * @author Vlad Mihalcea
+ */
+public class ValuesListCteBuilder {
+
+	private final String tableName;
+
+	private final String[] columns;
+
+	private final List<Object[]> ids;
+
+	private StringBuilder statementBuilder;
+
+	public ValuesListCteBuilder(
+			String tableName,
+			String[] columns,
+			List<Object[]> ids) {
+		this.tableName = tableName;
+		this.columns = columns;
+		this.ids = ids;
+
+		this.statementBuilder = buildStatement();
+	}
+
+	public String toStatement(String statement) {
+		return statementBuilder + statement;
+	}
+
+	public List<Object[]> getIds() {
+		return ids;
+	}
+
+	private StringBuilder buildStatement() {
+		StringBuilder buffer = new StringBuilder();
+
+		StringBuilder columnsBuffer = new StringBuilder();
+
+		for ( int i = 0; i < columns.length; i++ ) {
+			columnsBuffer.append( columns[i] );
+			if ( i < columns.length - 1 ) {
+				columnsBuffer.append( "," );
+			}
+		}
```

Comment:
Is the block above not the same as `String columnsBuffer = String.join(",", columns)`? At the same time, `columnNames` or `joinedColumnNames` might be a better name.